R/estimate_uncertainty.R

```diff
@@ -49,32 +57,39 @@
 #' retro_rts <- construct_triangles(trunc_rts)
 #'
 #' retro_nowcasts <- fill_triangles(retro_rts, n = 5)
+#' # Estimate dispersion parameters using default (negative binomial error
+#' # model on the sums
 #' disp_params <- estimate_uncertainty(
 #'   pt_nowcast_matrices = retro_nowcasts,
 #'   trunc_reporting_triangles = trunc_rts,
 #'   retro_reporting_triangles = retro_rts,
-#'   n = 2
 #' )
 #' disp_params
 #'
-#' # Estimate dispersion parameters from rolling sum
+#' # Estimate dispersion parameters from rolling mean with a normal error model
 #' disp_params_agg <- estimate_uncertainty(
 #'   pt_nowcast_matrices = retro_nowcasts,
 #'   trunc_reporting_triangles = trunc_rts,
 #'   retro_reporting_triangles = retro_rts,
 #'   n = 2,
-#'   fun_to_aggregate = sum,
-#'   k = 2
+#'   error_model = .fit_distrib,
+#'   error_args = list(observation_model = "normal"
+#'   aggregator = zoo::rollmean,
+#'   aggregator_args = list(k = 1, align = "right")
 #' )
 #' disp_params_agg
 estimate_uncertainty <- function(
     pt_nowcast_matrices,
     trunc_reporting_triangles,
     retro_reporting_triangles,
     n = length(pt_nowcast_matrices),
-    fun_to_aggregate = sum,
-    k = 1) {
-  .validate_aggregation_function(fun_to_aggregate)
+    error_model = .fit_distrib,
+    error_args = list(observation_model = "negative binomial"),
```

Comment:
I think the function above should have this as a arg via a partial function approach so we should't need this argument.

Reply:
> no function(...) fit_dist(..., option = blah blah)

I am not following this